src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -720,12 +732,17 @@ internal void FinishOperationSyncSuccess(int bytesTransferred, SocketFlags flags
                             catch (ObjectDisposedException) { }
                         }
 
+                        SocketsTelemetry.Log.ConnectStop();
+
                         // Mark socket connected.
                         _currentSocket!.SetToConnected();
                         _connectSocket = _currentSocket;
                     }
                     else
                     {
+                        SocketsTelemetry.Log.ConnectFailed();
+                        SocketsTelemetry.Log.ConnectStop();
```

Comment:
Why have the Stop in both branches rather than just having it once after the else?

Reply:
Initially, I aimed to call Stop as soon as it possible. However, now Failed calls Stop, so there is only one call on this line.